easybuild/easyblocks/o/openssl.py

```diff
@@ -57,15 +59,24 @@ def sanity_check_step(self):
         for libdir_cand in ['lib', 'lib64']:
             if os.path.exists(os.path.join(self.installdir, libdir_cand)):
                 libdir = libdir_cand
+
         if libdir is None:
             raise EasyBuildError("Failed to determine library directory.")
 
-        custom_paths = {
-            'files': [os.path.join(libdir, x) for x in ['libcrypto.a', 'libcrypto.so', 'libcrypto.so.1.0.0',
-                                                        'libssl.a', 'libssl.so', 'libssl.so.1.0.0']] +
-                     ['bin/openssl'],
-            'dirs': [os.path.join(libdir, 'engines')],
-        }
+        if LooseVersion(self.version) < LooseVersion("1.1"):
+            custom_paths = {
+                'files': [os.path.join(libdir, x) for x in ['libcrypto.a', 'libcrypto.so', 'libcrypto.so.1.0.0',
+                                                            'libssl.a', 'libssl.so', 'libssl.so.1.0.0']] +
+                         ['bin/openssl'],
+                'dirs': [os.path.join(libdir, 'engines')],
+            }
+        else:
+            custom_paths = {
+                'files': [os.path.join(libdir, x) for x in ['libcrypto.a', 'libcrypto.so', 'libcrypto.so.1.1',
+                                                            'libssl.a', 'libssl.so', 'libssl.so.1.1']] +
+                         ['bin/openssl'],
+                'dirs': [os.path.join(libdir, 'engines-1.1')],
```

Comment:
@vanzod I think it makes sense to avoid duplication, by fleshing out the common stuff, something like:

```python
custom_paths = {
    'files': [os.path.join(libdir, x) for x in ['libcrypto.a', 'libcrypto.so', 'libssl.a', 'libssl.so']] +
             ['bin/openssl'],
    'dirs': [],
}

if LooseVersion(self.version) < LooseVersion("1.1"):
    custom_paths['files'].extend([os.path.join(libdir, 'libcrypto.so.1.0.0'), os.path.join(libdir, 'libssl.so.1.0.0')])
    custom_paths['dirs'].append(os.path.join(libdir, 'engines'))
else:
    custom_paths['files'].extend([os.path.join(libdir, 'libcrypto.so.1.1'), os.path.join(libdir, 'libssl.so.1.1')])
    custom_paths['dirs'].append(os.path.join(libdir, 'engines-1.1'))
```